### PR TITLE
Add code tour for entity classes

### PR DESCRIPTION
**Pull request by MilesHolland**
# Description

Adds a code tour describing the basics of AML entity classes. Includes a couple TODOs that are references to other tours that don't exist yet.

# All SDK Contribution checklist:
- [x] **The pull request does not introduce [breaking changes]**
- [x] **CHANGELOG is updated for new features, bug fixes or other significant changes.**
- [x] **I have read the [contribution guidelines](https://github.com/Azure/azure-sdk-for-python/blob/main/CONTRIBUTING.md).**

## General Guidelines and Best Practices
- [x] Title of the pull request is clear and informative.
- [x] There are a small number of commits, each of which have an informative message. This means that previously merged commits do not appear in the history of the PR. For more information on cleaning up the commits in your PR, [see this page](https://github.com/Azure/azure-powershell/blob/master/documentation/development-docs/cleaning-up-commits.md).

### CODE REVIEW

File: sdk/ml/azure-ai-ml/.tours/entity_classes_purpose_and_examples.tour [hunk 1]
```diff
@@ -0,0 +1,85 @@
...
+        "end": {
+          "line": 32,
+          "character": 25
+        }
+      },
+      "title": "Example Operation"
+    },
+    {
+      "file": "sdk/ml/azure-ai-ml/azure/ai/ml/entities/_schedule/schedule.py",
+      "description": "Another important feature of entity classes is their ability to be loaded to and from YAML. This is largely achieved by a separete set of classes call schema classes, but some boilerplate is still needed in the corresponding entity classes to achieve this, namely the `_load` and `dump` functions. Though these functions are mostly boilerplate, but you should still be aware of them. They sometimes contain custom logic to help convert between an entity class and its corresponding schema class if they differ.",
```
Comment (diondrapeck) [inline sdk/ml/azure-ai-ml/.tours/entity_classes_purpose_and_examples.tour:63]: nit for typo on "separate"

File: sdk/ml/azure-ai-ml/.tours/entity_classes_purpose_and_examples.tour [hunk 2]
```diff
@@ -0,0 +1,85 @@
...
+    {
+      "file": "sdk/ml/azure-ai-ml/azure/ai/ml/entities/_schedule/schedule.py",
+      "description": "This is the opposite of the '_from_rest_object' function. It produces a corresponding API-compatible object from the current entity.\n\nNow, with these two functions in mind, let's take a look at how and where they are used.",
+      "line": 248,
+      "title": "Common Function: _to_rest_object"
+    },
+    {
+      "directory": "sdk/ml/azure-ai-ml/azure/ai/ml/operations",
+      "description": "The API conversion steps are generally used by the files in the (repo)/sdk/ml/azure-ai-ml/azure/ai/ml/operations directory. The \\<resource\\>_operations files in this directory contain the actual logic to make and process requests from the Azure API.",
+      "title": "Related DIrectory: Operations"
```
Comment (diondrapeck) [inline sdk/ml/azure-ai-ml/.tours/entity_classes_purpose_and_examples.tour:37]: nit for typo, lowercase the "I" in Directory

File: sdk/ml/azure-ai-ml/.tours/entity_classes_purpose_and_examples.tour [hunk 3]
```diff
@@ -0,0 +1,85 @@
...
...y classes is their ability to convert themselves to and from their corresponding autorest representations. Take a look at the function called '_from_rest_object'. Notice the 'RestSchedule' typehint for the input? That class is the autorest equivalent of a JobSchedule, which this function consumes in order to produce a JobSchedule entity class.\n\nAt a glance, this might seem like a tedious level of redirection - after all, why can we not just use the Autorest classes directly? There are multiple reasons for that:\n\n - Implementing our own entity classes gives us much a more stable user-facing API, once which isn't liable to shift if other teams modify the API's we're leveraging.\n - Sometimes, we explicitly want to deviate from the API representation of a resource to achieve various user-friendly tradeoffs. Examples include hiding in-development features that we don't want users easily playing with, simplifying the representation of features to prevent users from easily mis-configuring resources, or simply hiding the awkward structure of the API object.\n \n For more information about the underlying API respresentation. Check out the code tour titled \"TODO WRITE AUTOREST TOUR\"",
```
Comment (diondrapeck) [inline sdk/ml/azure-ai-ml/.tours/entity_classes_purpose_and_examples.tour:24]: I really like this explanation

**Review by diondrapeck — CHANGES_REQUESTED** [review]

File: sdk/ml/azure-ai-ml/.tours/entity_classes_purpose_and_examples.tour (hunk 3 above)
Comment (MilesHolland) [reply sdk/ml/azure-ai-ml/.tours/entity_classes_purpose_and_examples.tour:24]: Thanks!

**Review by diondrapeck — APPROVED** [review]

**Pull request merged**